packages/js/product-editor/src/hooks/use-product-entity-prop.ts

```diff
@@ -3,34 +3,36 @@
  */
 import { useEntityProp } from '@wordpress/core-data';
 
-interface Metadata {
+interface Metadata< T > {
 	id?: number;
 	key: string;
-	value: string;
+	value: T;
 }
 
-function useProductEntityProp(
-	property: string
-): [ string, ( value: string ) => void ] {
+function useProductEntityProp< T >(
+	property: string,
+	fallbackValue: T
+): [ T, ( value: T ) => void ] {
 	const isMeta = property.startsWith( 'meta_data.' );
 	const metaPropertyKey = property.replace( 'meta_data.', '' );
 
-	const [ entityPropValue, setEntityPropValue ] = useEntityProp< string >(
+	const [ entityPropValue, setEntityPropValue ] = useEntityProp< T >(
 		'postType',
 		'product',
```

Comment:
Could we pass the postType into this hook as well, in order to still support variations for example?